contactsummary/contactsummary.php

```diff
@@ -127,9 +127,14 @@ function contactsummary_civicrm_alterSettingsFolders(&$metaDataFolders = NULL) {
  */
 function contactsummary_civicrm_pageRun($page) {
   if ($page instanceof CRM_Contact_Page_View_Summary) {
-    CRM_Core_Resources::singleton()->addSetting(array(
-      'tabSettings' => array('active' => CRM_Utils_Request::retrieve('selectedChild', 'String', $this, FALSE, 'contactsummary')),
-    ));
+    CRM_Core_Resources::singleton()
+      ->addSetting(array(
+        'tabSettings' => array('active' => CRM_Utils_Request::retrieve('selectedChild', 'String', $this, FALSE, 'contactsummary')),
+      ))
+      ->addVars('leaveAndAbsences', [
```

Comment:
Not sure if it's even possible to avoid this, but this hard coupling between the extensions is pretty bad

Reply:
Agreed. Some of these need to go into an API endpoint (logged in contact id, upload token, etc.)

Reply:
@AkA84 @reneolivo Just reminding that we have a ticket for such API: PCHR-2465. We definitely should implement it and use it instead

Reply:
That's really nice @davialexandre!!! The only issue is that the shared settings is used to load template files:

Component:
```js
templateUrl: ['shared-settings', function (sharedSettings) {
  return sharedSettings.sharedPathTpl + 'components/leave-widget/leave-widget.html';
}]
```

Shared Settings:
```js
sharedPathTpl: CRM.vars.leaveAndAbsences.baseURL + '/views/shared/',
```

And templateUrl nor template don't seem to work with promises. (was checking). Maybe there could be a workaround, but this needs more investigation.